app/controllers/api/v1/orders_controller.rb

```diff
@@ -3,6 +3,7 @@ module V1
     class OrdersController < ApplicationController
       include Api::V1::Mixins::IndexMixin
       before_action :read_access_check, :only => %i[show]
+      before_action :service_offering_check, :only => %i[submit_order]
```

Comment:
@eclarizio How are the before_actions ordered, is it possible that the service_offering_check runs before the RBAC checks.

Reply:
Where are the RBAC checks being ran? The `read_access_check` here is only run on the `show` action, unless you're talking about the `around_filter` in `ApplicationController` that is checking entitlements? Is that RBAC?

Reply:
@eclarizio my question was a general question when you have a bunch of before_action's in which order do they run. But it seems like submit_order doesn't have any RBAC checks which  seems incorrect, our RBAC till now has been focused on the UI, if we can hide it from the user in the UI the RBAC works. If the API gets called directly with a submit order then there is no RBAC check on the Portfolio Item and the Portfolio.

Reply:
@mkanoor I think it depends on the structure.

For example, like this:
```ruby
class MyController < ApplicationController
  before_action :action1
  before_action :action2
end
```

From what I can tell from googling around, it appears as though you cannot guarantee that the order is going to be `action1, action2`, but if you structure it like this:
```ruby
class MyController < ApplicationController
  before_action :action1, :action2
end
```

Then the order is guaranteed. We could also do something like:
```ruby
class MyController < ApplicationController
  before_action :all_the_before_actions

  private

  def all_the_before_actions
    action1
    action2
  end
end
```

This would make extra sure that the ordering is the way we intend it to be, and might be the way we go. It would then make sense for the `service_offering_check` to simply call the appropriate RBAC check before it even does its own check.

I do think adding the RBAC check is outside the scope of this PR, though.